pkg/provider/vault/vault.go

```diff
@@ -186,12 +188,13 @@ func (v *client) readSecret(ctx context.Context, path, version string) (map[stri
 		// Vault KV2 has data embedded within sub-field
 		// reference - https://www.vaultproject.io/api/secret/kv/kv-v2#read-secret-version
 		dataInt, ok := vaultSecret.Data["data"]
+
 		if !ok {
-			return nil, errors.New(errVaultData)
+			return nil, fmt.Errorf(errDataField, vaultSecret.Data)
```

Comment:
While this is good for debugging i'd prefer not to log secret values (in line 197, 209 below too).

ref: https://cheatsheetseries.owasp.org/cheatsheets/Logging_Cheat_Sheet.html#data-to-exclude

Reply:
Oh, good point. I scrolled through and missed this being the actual data also, which we should never log. Could be ok to log ids or some other stuff, but even then could be problematic.